### PR TITLE
ESQL: Improve verifier error for incorrect agg declaration

### LINKED ISSUE #100634: Support aggregations across constants
### Description

In ESQL it is possible to have aggregations working on a constant as long as it comes from a source, however pushing it into the aggregation fails.

This works: `row a = 1 |  stats m = min(a)`  but the query `stats m = min(1)` does not.
We've been looking at adding a dedicated rule for optimizing this ( #100182) however the issue is rather related to folding aggregations.

A somewhat variant of this is folding the aggregation when the argument is the group key itself:
`stats mi = min(k),  ma = max(k), c = count(k), a = avg(k) by k`  which is the same as `stats by k | keep k as mi, k as ma, k as c, k as a, k`

- [x] Implement for aggs where `agg(const)` can be expressed using a multi-value function as `mv_...(const)` https://github.com/elastic/elasticsearch/pull/105454 https://github.com/elastic/elasticsearch/pull/107414
- [ ] Remove corresponding awaitsFixes from tests, and enable commented-out code
- [ ] Implement for aggs without corresponding multi-value functions ([`PERCENTILE`](https://github.com/elastic/elasticsearch/issues/111591), [`MEDIAN_ABSOLUTE_DEVIATION`](https://github.com/elastic/elasticsearch/issues/111590) [`ST_CENTROID_AGG`](https://github.com/elastic/elasticsearch/issues/108215))
- [ ] Enable PropagateEvalFoldables for aggregates

### LINKED ISSUE #100641: ESQL: Better error message for agg functions without brackets
### Description

`stats x = count`  throws error `unknown column [count]` which is confusing since stats does not allow aliasing for existing columns but only points to aggregate functions.
Hence why the message should be `incorrect function declaration count, you forgot to add the parantheisis`. 
Same approach when dealing with a typo: `stats x = cont(*)` - `Unkown aggregate function cont - did you mean count()?`

### PR DESCRIPTION
**Pull request by costin**
Add specific check for aliased function declarations in stats

Fix #100641

### CODE REVIEW

**Review by luigidellaquila — APPROVED** [review]
LGTM even though in some cases it could be misleading, eg `... | stats a = foo` will result in `incomplete aggregate function declaration; add parenthesis to [foo]`, and when the user tries to add the parentheses `... | stats a = foo()` he will just get another error.
Maybe we could be more generic and just say `invalid stats declaration; [{}] is not an aggregate function`. 

As a future evolution, we could also use the FunctionRegistry to suggest alternatives, like `did you mean xyz()?`

**Review by astefan — COMMENTED** [review]
Imho, I believe we are too picky with the error messages improvements. Yes, we can do better in some situations but there will always be users who will complain about the message.
And @luigidellaquila's example is very good in pointing out another of these edge situations where we could do better.

Another one is `stats c = avg(`, where one could argue that the language should have a better error message than `no viable alternative at input 'avg('` (ie "why isn't the language telling me that I should add another parenthesis?").

In any case, the improvement in this PR is welcomed, BUT I suggest incorporating the fix that @luigidellaquila mentioned in this PR not in a new one.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java [hunk 1]
```diff
@@ -89,6 +91,19 @@ Collection<Failure> verify(LogicalPlan plan) {
             else if (p.resolved()) {
                 return;
             }
+            // handle aggregate first to disambiguate between missing fields or incorrect function declaration
+            if (p instanceof Aggregate aggregate) {
+                for (NamedExpression agg : aggregate.aggregates()) {
```
Comment (alex-spies) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java:96]: Should we maybe consolidate this with the verification happening below in lines 125/140? Now we're verifying aggs in two places.

**Review by alex-spies — APPROVED** [review]

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java (hunk 1 above)
Comment (costin) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java:97]: There are different types of verifications - the first one is around unknown fields the other around resolved fields but incorrectly declared.

**Pull request merged**